### PR TITLE
Redirecting routes

### PR DESCRIPTION
**Pull request by uo271580**
Hey i made the redirecting routes #136 

- When a user is not logged it won't be able to access the dashboard or the checkout
- When a user is logged it won't be able to access log in or sign up pages
- User without permissions won't be able to access upload and delete product
- Every incorrect route typed will redirect directly to home page

Check everything works fine

### CODE REVIEW

File: webapp/src/components/dashboard/products/DeleteProduct.tsx [hunk 1]
```diff
@@ -95,127 +94,135 @@ export default function DeleteProduct(props: DeleteProductProps): JSX.Element {
...
-            ))}
-          </TextField>
-
-          <Stack
-            direction="row"
-            spacing={2}
-            justifyContent="space-evenly"
-            alignItems="stretch"
+
+  if (
```
Comment (angelip2303) [inline webapp/src/components/dashboard/products/DeleteProduct.tsx:98]: Probably these lines could be refactored into a parent component where the children is UpdateProduct and DeleteProduct and you pass it as a chidlren.

File: webapp/src/components/dashboard/products/UploadProduct.tsx [hunk 2]
```diff
@@ -131,137 +130,144 @@ export default function UploadImage(props: UploadProductProps): JSX.Element {
...
-          sx={{ my: { xs: 3, md: 6 }, p: { xs: 2, md: 3 } }}
-        >
-          <h1 style={{ margin: 8 }}>Upload a product</h1>
-
-          <Stack
-            direction="row"
-            spacing={2}
-            justifyContent="space-evenly"
-            alignItems="stretch"
+  if (
```
Comment (angelip2303) [inline webapp/src/components/dashboard/products/UploadProduct.tsx:133]: Same here, you can just render UploadProduct and in a parent something like:

```suggestion
   if (localStorage.getItem("user.email") === null ||
        (localStorage.getItem("user.role") !== "admin" &&
         localStorage.getItem("user.role") !== "manager")
      return <Navigate to="/" />;
  else
      { props.children }
  )
```
Where props.children is this component: UploadProduct

**Review by angelip2303 — CHANGES_REQUESTED** [review]
There are some changes that I would made in order this to be perfect

File: webapp/src/components/dashboard/products/DeleteProduct.tsx (hunk 1 above)
Comment (uo271580) [reply webapp/src/components/dashboard/products/DeleteProduct.tsx:98]: I really don't get your point here. What do you mean?

**Pull request merged**